src/common/snippets/src/pass/hash.cpp

```diff
@@ -0,0 +1,453 @@
+        m_hash = hash_combine(hash_combine(m_hash, name), adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_attribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int64_t>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_attribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<uint64_t>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_attribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<float>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_attribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_attribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::shared_ptr<ov::Model>>& adapter) override {
+        ovfunction_2_hash(m_hash, *adapter.get());
+    }
+};
+OPENVINO_SUPPRESS_DEPRECATED_END
+
+std::unordered_map<ov::Node*, int> create_layer_ids(const ov::Model& model) {
+    std::unordered_map<ov::Node*, int> layer_ids;
+    int id = 0;
+    for (const auto& node : model.get_ordered_ops()) {
+        layer_ids[node.get()] = id++;
+    }
+    return layer_ids;
+}
+
+std::vector<Edge> create_edge_mapping(const std::unordered_map<ov::Node*, int>& layer_ids,
+                                            const ov::Model& model) {
+    std::vector<Edge> edges;
+    for (const auto& node : model.get_ordered_ops()) {
+        if (ov::op::util::is_parameter(node)) {
+            continue;
+        }
+
+        for (const auto& i : node->inputs()) {
+            auto source_output = i.get_source_output();
+            auto source_node = source_output.get_node();
+            auto current_node = i.get_node();
+
+            if (layer_ids.find(source_node) == layer_ids.end() || layer_ids.find(current_node) == layer_ids.end()) {
+                OPENVINO_THROW("Failed creat edge map in snippets hash generation");
+            }
+
+            Edge e{};
+            e.from_layer = layer_ids.find(source_node)->second;
+            e.from_port = static_cast<int>(source_node->get_input_size() + source_output.get_index());
+            e.to_layer = layer_ids.find(current_node)->second;
+            e.to_port = static_cast<int>(i.get_index());
+            edges.push_back(e);
+        }
+    }
+    std::sort(begin(edges), end(edges), [](const Edge& a, const Edge& b) -> bool {
+        return a.from_layer < b.from_layer;
+    });
+    return edges;
+}
+
+std::string get_precision_name(const ov::element::Type& elem_type) {
```

Comment:
`ov::element::Type` already has overrided `<<` operator  and even `hash()` method. Can we reuse smt instead?

Reply:
That's great! element::Type hash() is used. Thanks Dmitry.